src/main/java/com/ebay/myriad/scheduler/TaskFactory.java

```diff
@@ -63,6 +69,118 @@ private static String getFileName(String uri) {
+            } else {
+                String address = conf.get(YARN_RESOURCEMANAGER_WEBAPP_ADDRESS);
+                if (address == null || address.isEmpty()) {
+                    address = conf.get(YARN_RESOURCEMANAGER_HOSTNAME) + ":8088";
+                }
+                return "http://" + address + "/conf";
+            }
+        }
+
+        private Protos.CommandInfo getCommandInfo() {
+            MyriadExecutorConfiguration myriadExecutorConfiguration = cfg.getMyriadExecutorConfiguration();
+            CommandInfo.Builder commandInfo = CommandInfo.newBuilder();
+            if (myriadExecutorConfiguration.getNodeManagerUri().isPresent()) {
+                /*
+                 Overall this is messier than I'd like due to Mesos 1790. We can't let mesos untar the distribution,
+                 since it will change the permissions.  Instead we simply download the tarball and execute tar -xvpf.
+                 We also pull the config from the resource manager and put them in the conf dir.  This is also why we need
+                 frameworkSuperUser
+                */
+
+                //Too much work checking all posibilities and many are likely to fail somewhere later anyway
+                //Keep in mind this is a temp fix anyway
+                if (!(cfg.getFrameworkUser().isPresent() && cfg.getFrameworkSuperUser().isPresent())) {
+                    throw new RuntimeException("Trying to use remote distribution, but frameworkUser" +
+                            "and/or frameworkSuperUser not set!");
+                }
+
+                LOGGER.info("Using remote distribution");
+
+                String executorCmdPrefix = "export CAPSULE_CACHE_DIR=`pwd`;echo $CAPSULE_CACHE_DIR; " +
+                        "sudo -E -u " + cfg.getFrameworkUser().get() + " -H " +
+                        "java -Dcapsule.log=verbose -jar ";
+
+                String executorPathString = myriadExecutorConfiguration.getPath();
+                String nmURIString = myriadExecutorConfiguration.getNodeManagerUri().get();
+
+                //TODO(DarinJ) support other compression, as this is a temp fix for Mesos 1760 may not get to it.
+                //Extract tarball keeping permissions
+                String tarCmd = "sudo tar -zxpf " + getFileName(nmURIString);
+
+                /*
+                We need to current directory to be owned by framework users for capsuleExecutor to create directories
+                Can't do this if framework user not set, may still be ok esp. if frameworkUser==frameSuperUser, logging a
+                warning.
+                */
+                String chownCmd = "sudo chown " + cfg.getFrameworkUser().get() + " .";
+
+                //Throw the hadoop config where it can be picked by the NodeManager
+                String configCopyCmd = "cp conf " + cfg.getYarnEnvironment().get("YARN_HOME") +
+                        "/etc/hadoop/yarn-site.xml";
```

Comment:
Is 'conf' a file or directory? Do you really want it to be copied into 'yarn-site.xml'? If it is actually a file, why isn't it named something.xml? If it's a directory, did you mean to do a recursive copy into `$YARN_HOME/etc/hadoop` instead of into a directory named 'yarn-site.xml'?